### PR TITLE
[SOLID] Change if && to early return (process nested ifs)

### PR DESCRIPTION
**Pull request by dobryy**
Follow-up: https://github.com/rectorphp/rector/pull/4344#discussion_r502641921 and https://github.com/rectorphp/rector/pull/4344#discussion_r502642814
Related to: #4071

### CODE REVIEW

File: rules/sensio/src/TypeDeclaration/ReturnTypeDeclarationUpdater.php [hunk 1]
```diff
@@ -86,9 +86,13 @@ private function updatePhp(ClassMethod $classMethod, string $className): void
...
+            if ($returnTypeName === null) {
                 return;
             }
+            if (!is_a($returnTypeName, $className, true)) {
+                return;
+            }
+            return;
         }
 
         $classMethod->returnType = new FullyQualified($className);
```
Comment (dobryy) [inline rules/sensio/src/TypeDeclaration/ReturnTypeDeclarationUpdater.php]: @TomasVotruba this looks like an edge case that has two options to be solved:
1. Skip `if` if it contains `return;` (void) statement
2. Process if and replace `if`'s statements with `continue;` statement

Let me know what is the preferred solution in your opinion.

Comment (TomasVotruba) [reply rules/sensio/src/TypeDeclaration/ReturnTypeDeclarationUpdater.php]: I think 1. is right in this case.
What do you think?

Comment (dobryy) [reply rules/sensio/src/TypeDeclaration/ReturnTypeDeclarationUpdater.php]: For this particular rule option #1 makes more sense to me as well.

**Comment by dobryy** [comment]
@TomasVotruba If it looks fine to you, this PR is ready for merging.

**Review by TomasVotruba — APPROVED** [review]

**Comment by TomasVotruba** [comment]
Looks good to me, thank you

**Pull request merged**